doc/sphinx-guides/source/developers/coding-style.rst

```diff
@@ -2,26 +2,38 @@
 Coding Style
 ============
 
-Like all development teams, the `Dataverse developers at IQSS <https://dataverse.org/about>`_ have their habits and styles when it comes to writing code. Let's attempt to get on the same page. :)
-
-.. contents:: |toctitle|
-	:local:
-
 Java
 ----
 
 Formatting Code
 ~~~~~~~~~~~~~~~
 
-Tabs vs. Spaces
-^^^^^^^^^^^^^^^
+NetBeans
+^^^^^^^^
+
+We highly recommend using Netbeans as a text editor. This makes it significantly easier to commit code that is well formatted and type safe. By default, NetBeans will warn you about some things, but if you want better warnings and code formatting, download this##################TODO ADD FILE######################### zip file, and import it to your nNtBeans settings by going to options, import, and selecting the file. These settings will cause NetBeans to automatically format code that you've worked on whenever you save, point out many common problems in code, and make sure that all indentation uses spaces.
+
+Alternatively, you can set the warnings yourself. To do so, go to tools-> options-> editor -> hints, and turn on the following
+    - Braces
+    - Error Fixes
+    - JDK Migration Support
+    - Probable Bugs
+    - Standard Javac Warnings
+
 
-Don't use tabs. Use spaces.
+Not NetBeans
+^^^^^^^^^^^^
 
-Format Code You Changed with Netbeans
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you are not going to use NetBeans, here are code style tips in order to be consistent with the rest of the project.
+- Use 4 spaces for indentation (not tabs)
+- Make sure your code is type safe.
+- Make sure not to use raw classes
+- Use diamond interface when applicable
+- Use javadoc and comments to help explain what your code is doing
+- Try to make your code look roughly like the following sample
 
-As you probably gathered from the :doc:`dev-environment` section, IQSS has standardized on Netbeans. It is much appreciated when you format your code (but only the code you touched!) using the out-of-the-box Netbeans configuration. If you have created an entirely new Java class, you can just click Source -> Format. If you are adjusting code in an existing class, highlight the code you changed and then click Source -> Format. Keeping the "diff" in your pull requests small makes them easier to code review.
```

Comment:
The bit about "keeping the 'diff' in your pull requests small makes them easier to code review" is very important to me. I don't want a new developer to join the project and say, "Hey, the coding style pages says we use spaces, so I reformatted the entire code base to remove tabs. Here's a pull request." I want a new developer to focus on fixing an actual bug or improving the user experience for end users with a small "diff", a small pull request.

When working on code, you should only reformat the code you're working on. Likewise, you should only improve the type safety of the code you're working on. Eventually, we'll make the code consistent and of higher quality. It won't happen in one summer.